perfkitbenchmarker/linux_benchmarks/object_storage_service_benchmark.py

```diff
@@ -501,278 +433,245 @@ def _DistributionToBackendFormat(dist):
+                                       LATENCY_UNIT,
+                                       metadata)
       else:
-        logging.error('Exceeded max retry limit for removing the empty bucket')
-        raise BucketRemovalError('Failed to remove the bucket')
+        raise ValueError('Unexpected test outcome from OneByteRW api test: '
+                         '%s.' % raw_result)
+
+
+def SingleStreamThroughputBenchmark(results, metadata, vm, command_builder,
+                                    service, bucket_name, regional_bucket_name):
+  single_stream_throughput_cmd = command_builder.BuildCommand([
+      '--bucket=%s' % bucket_name,
+      '--scenario=SingleStreamThroughput'])
+
+  _, raw_result = vm.RemoteCommand(single_stream_throughput_cmd)
+  logging.info('SingleStreamThroughput raw result is %s', raw_result)
+
+  for up_and_down in ['upload', 'download']:
+    search_string = 'Single stream %s throughput in Bps: (.*)' % up_and_down
+    result_string = re.findall(search_string, raw_result)
+    sample_name = SINGLE_STREAM_THROUGHPUT % up_and_down
+
+    if len(result_string) > 0:
+      # Convert Bytes per second to Mega bits per second
+      # We use MB (10^6) to be consistent with network
+      # bandwidth convention.
+      result = json.loads(result_string[0])
+      for percentile in PERCENTILES_LIST:
+        results.append(sample.Sample(
+            ('%s %s') % (sample_name, percentile),
+            8 * float(result[percentile]) / 1000 / 1000,
+            THROUGHPUT_UNIT,
+            metadata))
+    else:
+      raise ValueError('Unexpected test outcome from '
+                       'SingleStreamThroughput api test: %s.' % raw_result)
+
+
+def ListConsistencyBenchmark(results, metadata, vm, command_builder,
+                             service, bucket_name, regional_bucket_name):
+  list_consistency_cmd = command_builder.BuildCommand([
+      '--bucket=%s' % bucket_name,
+      '--iterations=%d' % FLAGS.object_storage_list_consistency_iterations,
+      '--scenario=ListConsistency'])
+
+  _, raw_result = vm.RemoteCommand(list_consistency_cmd)
+  logging.info('ListConsistency raw result is %s', raw_result)
+
+  for scenario in LIST_CONSISTENCY_SCENARIOS:
+    metric_name = '%s %s' % (scenario, LIST_CONSISTENCY_PERCENTAGE)
+    search_string = '%s: (.*)' % metric_name
+    result_string = re.findall(search_string, raw_result)
+    if len(result_string) > 0:
+      results.append(sample.Sample(metric_name,
+                                   (float)(result_string[0]),
+                                   NA_UNIT,
+                                   metadata))
+    else:
+      raise ValueError(
+          'Cannot get percentage from ListConsistency test.')
+
+    # Parse the list inconsistency window if there is any.
+    metric_name = '%s %s' % (scenario, LIST_INCONSISTENCY_WINDOW)
+    search_string = '%s: (.*)' % metric_name
+    result_string = re.findall(search_string, raw_result)
+    if len(result_string) > 0:
```

Comment:
I think this and the following `if`statement must always be taken, because if `len(result_string) == 0` then we already raised an exception above. So I removed them.